components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIProviderHostObject.java

```diff
@@ -4770,11 +4771,11 @@ private static String sendHttpHEADRequest(String urlVal) {
 
         try {
             HttpResponse httpResponse = client.execute(head);
-            int statusCode = httpResponse.getStatusLine().getStatusCode();
+            String statusCode = String.valueOf(httpResponse.getStatusLine().getStatusCode());
 
-            //If the endpoint doesn't support HTTP HEAD or if status code is < 400
-            if (statusCode == 405 || statusCode / 100 < 4) {
-                if (log.isDebugEnabled() && statusCode == 405) {
+            //If the endpoint doesn't match the regex which specify invalid status codes, it will return success.
+            if (!statusCode.matches(invalidStatusCodesRegex)) {
+                if (log.isDebugEnabled() && statusCode.equals("405")) {
```

Comment:
use "405".equals and better move the status code to  a constant. May be you can take this from standard available class for http status codes